### PR TITLE
Fix deploy on AKS

### PR DESCRIPTION
**Pull request by meldafrawi**
create DS to add containerd config to allow http for local registry
fix https://github.com/acorn-io/acorn/issues/670

Signed-off-by: mohamed.eldafrawi <mohamed@acorn.io>

### CODE REVIEW

File: pkg/build/buildkit/template.go [hunk 1]
```diff
@@ -231,3 +256,58 @@ func objects(namespace, buildKitImage, registryImage string) []client.Object {
...
+						Containers: []corev1.Container{
+							{
+								Name: system.ContainerdConfigPathName,
+								Env: []corev1.EnvVar{
+									{
+										Name:  "REGISTRY_SERVICE_NODEPORT",
+										Value: registryServiceNodePort,
+									},
+								},
+								Image: "meldafrawi/containerd-registry-config:latest",
```
[the remark this answers is not in the document]
Comment (meldafrawi) [reply pkg/build/buildkit/template.go]: done.

File: pkg/build/buildkit/deploy.go [hunk 2]
```diff
@@ -49,6 +70,12 @@ func GetBuildkitPod(ctx context.Context, client client.WithWatch) (int, *corev1.
 		if err != nil {
 			return 0, nil, err
 		}
+
+		err1 := applyContainerdConfigPath(ctx)
```
Comment (ibuildthecloud) [inline pkg/build/buildkit/deploy.go:74]: Can this logic be in applyObjects?

File: scripts/ds-containerd-config-path-entry [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+# check if config_path is defined for containerd registry plugin
+CONTAINERD_REGISTRY_CONFIG_PATH_DEFINED=$(sed -n '/^.*\[plugins\.\"io\.containerd\..*\"\.registry\]/,/^.*\[plugins\..*\]/p' "${CONTAINERD_CONFIG_DIR}" | grep -oE "^.*config_path += +[\"/A-Za-z\.]*" | grep -v "^\ *#")
+
+if [ "${CONTAINERD_REGISTRY_CONFIG_PATH_DEFINED}" ]; then
+	CONTAINERD_REGISTRY_CONFIG_PATH=$(echo ${CONTAINERD_REGISTRY_CONFIG_PATH_DEFINED} | tr -d '\"\|\ ' | awk -F "=" '{print $2}')
+	echo "Info: containerd registry plugin config_path: ${CONTAINERD_REGISTRY_CONFIG_PATH}"
+
+	# check REGISTRY_SERVICE_NODEPORT env variable is not empty
+	if [ ${REGISTRY_SERVICE_NODEPORT} ]; then
```
Comment (ibuildthecloud) [inline scripts/ds-containerd-config-path-entry:13]: Should this be quoted? I feel like your doing a bash trick I'm not familiar with.

File: pkg/build/buildkit/deploy.go [hunk 4]
```diff
@@ -27,14 +27,35 @@ func Exists(ctx context.Context, c client.Client) (bool, error) {
 	} else if err != nil {
 		return false, err
 	}
+
+	// check if the controller is running outside of a deployment
+	err = checkControllerDeployment(ctx, c)
```
Comment (cjellick) [inline pkg/build/buildkit/deploy.go:32]: I still think checking for the env var being set is a cleaner approach and abstracts you away from having to know about "how" the controller was deployed, but i want to get this in, so merging as is.

**Review by cjellick — APPROVED** [review]

**Pull request merged**